### PR TITLE
Implement HW3.

### PR DESCRIPTION
**Pull request by naztekinalp**

### CODE REVIEW

File: HW3/Author.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package HW3;
+
+public class Author {
+    int ID;
```
Comment (drevalanya) [inline HW3/Author.java]: all class fields and method attributes should start from lowercase

File: HW3/Main.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                new Book(1111, "Hannibal Lecter", new Author(1755, "Ilyas", "Akin"), new Publisher(3333, "Ekrem", "Rize"), 1269, false),
+                new Book(98642, "Usdagul", new Author(99666, "Taylor", "Swift"), new Publisher(3167, "Abdulhamit", "Van"), 1999, false),
+
+        };
+
+        for (int i = 0; i < books.length; i++) {
+            if (books[i].Paperback == false) {
+                System.out.println(books[i].ISBN);
+            }
+
```
Comment (drevalanya) [inline HW3/Main.java:20]: the 3rd point of the hw isn't fulfilled - program must have a method to go throw the array and display books by one of criteria.
this should be another method, not only main.

File: HW3/Author.java (hunk 1 above)
Comment (naztekinalp) [reply HW3/Author.java]: updated in [5511bf4](https://github.com/naztekinalpp/NazTekinalp/pull/6/commits/5511bf40b8a9d54751805b39d256d7d91ea08ee8)

File: HW3/Main.java (hunk 2 above)
Comment (naztekinalp) [reply HW3/Main.java:20]: updated in [8f979b8](https://github.com/naztekinalpp/NazTekinalp/pull/6/commits/8f979b8fe03cf0d0cea8c042394ed0820f0dbe9b)

File: HW3/Main.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public static void main(String[] args) {
+        Book[] books = {
+                new Book(1234, "Bas", new Author(8132, "James", "Hetfield"), new Publisher(9111, "Kemal", "Izmir"), 1920, true),
+                new Book(5364563, "Dijsktra", new Author(6456462, "Naz", "Tekinalp"), new Publisher(7474, "Recaizade", "Ankara"), 2023, true),
+                new Book(232423434, "Red Orange", new Author(233, "Mick", "Jagger"), new Publisher(42424, "Mahmut", "Istanbul"), 1071, true),
+                new Book(1111, "Hannibal Lecter", new Author(1755, "Ilyas", "Akin"), new Publisher(3333, "Ekrem", "Rize"), 1269, false),
+                new Book(98642, "Usdagul", new Author(99666, "Taylor", "Swift"), new Publisher(3167, "Abdulhamit", "Van"), 1999, false),
+
+        };
+        filterByISBN(books, 1234);
```
Comment (drevalanya) [inline HW3/Main.java]: useless call of the function, if next line you are calling it again

File: HW3/Main.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    public static ArrayList<Book> filterByISBN(Book[] books, int value) {
+        ArrayList<Book> booksToReturn = new ArrayList<Book>();
+
+        for (int i = 0; i < books.length; i++) {
+            if (books[i].isbn == value) {
+                booksToReturn.add(0, books[i]);
+            }
+        }
```
Comment (drevalanya) [inline HW3/Main.java:25]: it is possible to simplify this loop 
for (Book book : books) {
      if (book.isbn == value) {
                 booksToReturn.add(0, book);

File: HW3/Book.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    Book(int isbn, String name, Author author, Publisher publisher) {
+        this.isbn = isbn;
+        this.name = name;
+        this.author = author;
+        this.publisher = publisher;
+    }
+
+    public void setISBN(int ISBN) {
```
Comment (drevalanya) [inline HW3/Book.java]: incorrect name of attribute

Comment (naztekinalp) [reply HW3/Book.java]: Fixed in [f99708b](https://github.com/naztekinalpp/NazTekinalp/pull/6/commits/f99708b9e8d4241c017542d819fe8ef804d2d574)

File: HW3/Main.java (hunk 4 above)
Comment (naztekinalp) [reply HW3/Main.java:25]: fixed in [d78ff59](https://github.com/naztekinalpp/NazTekinalp/pull/6/commits/d78ff5930ce2f6e2e1092e1f8024f216772a960c)

File: HW3/Main.java (hunk 3 above)
Comment (naztekinalp) [reply HW3/Main.java]: updated [6da4c53](https://github.com/naztekinalpp/NazTekinalp/pull/6/commits/6da4c5335b01b91dc2f148bcded59f7f3132b0e4)

**Review by drevalanya — APPROVED** [review]

**Pull request merged**